cmd/tracee-ebpf/capabilities.go

```diff
@@ -73,16 +74,47 @@ func getCapabilitiesRequiredByTraceeEvents(cfg *tracee.Config) []cap.Value {
 	return removeDupCaps(caps)
 }
 
+// Retrieve the value of the kernel parameter perf_event_paranoid
+func getKernelPerfEventParanoidValue() int {
+	// Read the kernel parameter perf_event_paranoid
+	value, err := os.ReadFile("/proc/sys/kernel/perf_event_paranoid")
+	// If file read failed, returns the most restrictive value
+	if err != nil {
+		return 4
+	}
+	// Strip the newline character
+	value = value[:len(value)-1]
+	// Convert the value to an integer
+	intVal, err := strconv.ParseInt(string(value), 0, 16)
+	// If conversion failed, returns the most restrictive value
+	if err != nil {
+		return 4
+	}
+	return int(intVal)
+}
+
 // Get all capabilities required for eBPF usage (including perf buffers maps management)
 func getCapabilitiesRequiredByEBPF(OSInfo KernelVersionInfo) ([]cap.Value, error) {
 	// In kernel 5.8, CAP_BPF and CAP_PERFMON capabilities were introduced in order to replace CAP_SYS_ADMIN when
 	// loading eBPF programs.
-	// For some reasons, some distributions using new kernels still need CAP_SYS_ADMIN,
-	// so tracee still use it instead of the new capabilities.
-	caps := []cap.Value{
-		cap.IPC_LOCK,
-		cap.SYS_RESOURCE,
-		cap.SYS_ADMIN,
+	kernelParanoidValue := getKernelPerfEventParanoidValue()
```

Comment:
Hum, thats a nice catch =). I would add comment from `kernel/events/core.c` as a comment here (or in that function):

```
/*
 * perf event paranoia level:
 *  -1 - not paranoid at all
 *   0 - disallow raw tracepoint access for unpriv
 *   1 - disallow cpu events for unpriv
 *   2 - disallow kernel profiling for unpriv
 *   4 - disallow all unpriv perf event use
 */
```

or something similar.